File: 1651-shuffle-string/1651-shuffle-string.java
class Solution {
    public String restoreString(String s, int[] indices) {
        char[] answer = new char[indices.length];
        for (int i = 0; i < indices.length; i ++) {
            answer[indices[i]] = s.charAt(i);
        }
        return new String(answer);
    }
}

File: 1651-shuffle-string/1651-shuffle-string.js
/**
 * @param {string} s
 * @param {number[]} indices
 * @return {string}
 */
var restoreString = function(s, indices) {
    let answer = [];
    for (let i = 0; i < indices.length; i ++) {
        answer[indices[i]] = s[i];
    }
    return answer.join("");
};